aiida/orm/implementation/django/comments.py

```diff
@@ -117,16 +120,81 @@ def create(self, node, user, content=None, **kwargs):
         """
         return DjangoComment(self.backend, node, user, content, **kwargs)
 
-    def delete(self, comment):
+    def delete(self, comment_id):
         """
         Remove a Comment from the collection with the given id
 
-        :param comment: the id of the comment to delete
+        :param comment_id: the id of the comment to delete
+        :type comment_id: int
+
+        :raises TypeError: if ``comment_id`` is not an `int`
+        :raises `~aiida.common.exceptions.NotExistent`: if Comment with ID ``comment_id`` is not found
         """
-        # pylint: disable=no-name-in-module,import-error
-        from django.core.exceptions import ObjectDoesNotExist
-        assert comment is not None
+        # Check
+        if not isinstance(comment_id, int):
+            raise TypeError("comment_id must be an int")
+
+        # Delete
         try:
-            models.DbComment.objects.get(id=comment).delete()
+            with transaction.atomic():
+                models.DbComment.objects.get(id=comment_id).delete()
         except ObjectDoesNotExist:
-            raise exceptions.NotExistent("Comment with id '{}' not found".format(comment))
+            raise exceptions.NotExistent("Comment with id '{}' not found".format(comment_id))
+
+    def delete_all(self):
+        """Delete all Comment entries."""
+        with transaction.atomic():
+            models.DbComment.objects.all().delete()
+
+    # pylint: disable=too-many-branches,fixme
+    def delete_many(self, filters):
+        """
+        Delete Comments based on `filters`.
+        If `filters` is an empty dict, raise ValidationError.
+        If at least one entity is found, that entity is deleted, nothing is raised.
+
+        :param filters: similar to QueryBuilder filter
+        :type filters: dict
+
+        :raises TypeError: if ``filters`` is not a `dict`
+        :raises `~aiida.common.exceptions.ValidationError`: if ``filters`` is empty
+        :raises `~aiida.common.exceptions.NotExistent`: if no Comments are found
+        """
+        from aiida.orm import Comment, QueryBuilder
+        from aiida.orm.implementation.utils import get_filter_values, ignore_qb_operators
+
+        # Checks
+        if not isinstance(filters, dict):
+            raise TypeError("filters must be a dictionary")
+        if not filters:
+            raise exceptions.ValidationError("filters must not be empty")
+
+        # Initialization
+        not_found_entities = {}
+        column_mapping = {'node_id': 'dbnode_id'}
+        unique_keys = ['id', 'uuid']
+
+        # Special case for unique key values returning single entities
+        for key in unique_keys:
```

Comment:
I am a bit befuddled by this logic here. What is it you are trying to do here? Why are you manipulating the filters passed by the user in the arguments?

Reply:
Since this comment reiterates the general comment of this review, I'll answer here.
This logic is quite convoluted, yes. Even though it has gone through some iterations. Its point is to let the user use QueryBuilder (qb) filters, along with the operators available there. So I am essentially trying to parse the filter. However, I see now that I am, indeed, not doing it properly, since the `ignore_qb_operators` utility function basically does what it says on the box; ignore the operators, i.e., the operators have no effect.
Do we have a utility function to parse qb filters meaningfully, not creating SQL queries ....

Wait, I have just remembered (it's been a long time since I wrote this PR). This logic essentially shouldn't be here, since the filter should be passed directly to the qb to retrieve the PKs. However, there must have been some problem with this, since I did this logic.
Let me just do some further testing and I'll revise this accordingly...

Reply:
Ah! Now I remember. The problem is that if a user specifically requests to delete two Comments (or Logs) according to a unique identifier, then if one of the two entities do not exist, the QueryBuilder will not raise, but simply return the one entity that does exist. Hence, the user will never know that the non-existing entity could not be found (which may be annoying if it was a simple typo).

This (incorrect) logic was an attempt at providing the user with this information. However, since it doesn't handle the `or` and `and` operators correctly, it should either be removed or updated.